app/services/activities/import_from_csv.rb

```diff
@@ -284,19 +285,30 @@ def to_h
       end
 
       def convert_to_attributes
-        attributes = FIELDS.each_with_object({}) { |(attr_name, column_name), attrs|
+        attributes = fields.each_with_object({}) { |(attr_name, column_name), attrs|
           attrs[attr_name] = convert_to_attribute(attr_name, @row[column_name]) if field_should_be_converted?(column_name)
         }
 
-        attributes[:geography] = infer_geography(attributes)
-        attributes[:requires_additional_benefitting_countries] = (@row["Recipient Country"] && @row["Intended Beneficiaries"]).present?
-        attributes[:recipient_region] ||= inferred_region
-        attributes[:call_present] = (@row["Call open date"] && @row["Call close date"]).present?
-        attributes[:sector_category] = get_sector_category(attributes[:sector])
+        if @method == :create
+          attributes[:geography] = infer_geography(attributes)
+          attributes[:requires_additional_benefitting_countries] = (@row["Recipient Country"] && @row["Intended Beneficiaries"]).present?
+          attributes[:recipient_region] ||= inferred_region
+          attributes[:call_present] = (@row["Call open date"] && @row["Call close date"]).present?
+        end
+
+        attributes[:sector_category] = get_sector_category(attributes[:sector]) if attributes[:sector].present?
 
         attributes
       end
 
+      def fields
+        return FIELDS if @method == :create
+
+        columns_to_update = @row.to_h.reject { |_k, v| v.blank? }.keys
+        converter_keys = FIELDS.select { |_k, v| columns_to_update.include?(v) }.keys
+        FIELDS.slice(*converter_keys)
+      end
+
       def field_should_be_converted?(column_name)
         ALLOWED_BLANK_FIELDS.include?(column_name) || @row[column_name].present?
```

Comment:
I think the original purpose of this method was to allow the policy markers to be set to `not_assessed`. If we're not doing that anymore, is there any point to allowing some fields to be converted if they're blank?

Reply:
Also, a good shout. This will also make the logic a lot clearer. Thanks for noticing 👍

Reply:
@CristinaRO Thinking about this more, I think this is definitely a thing we should do, but won't be able to do until we set a default for the policy markers, which will come in a seperate feature.